lib/github-pages-health-check/domain.rb

```diff
@@ -243,6 +243,16 @@ def cname_to_github_user_domain?
         cname? && !cname_to_pages_dot_github_dot_com? && cname.pages_domain?
       end
 
+      # Check if the CNAME points to a Domain that points to pages
+      # e.g. CNAME -> Domain -> Pages
+      def cname_to_domain_to_pages?
```

Comment:
this function seems not check the cname point to A record which points to pages.

Does this allow the user input arbitrary cname?

Reply:
It should the main thing is that the `CNAME` doesn't already point to us then we run this check
https://github.com/github/pages-health-check/blob/8d25b396063aab1fdf5a2a9ec294287eddb0937b/lib/github-pages-health-check/domain.rb#L468
I have a test that checks if it points to it but let me add a test in the event that it doesn't

Reply:
I mainly mean if we want to only support `www.fontawesome.it`  => `fontawesome.it` => pagesxxx

or loosely support `blog.fontawesome.it` => `fontawesome.it` => `pagesxxxx`

hmmm perhaps fine?

Reply:
ohh I see let me double check on that!

Reply:
So it looks like we just want to support `www.fontawesome.it` => `fontawesome.it` => pages where we match the `host` after the `www`